core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -545,14 +545,19 @@ object ConsumerGroupCommand extends Logging {
         .map(partitionInfo => new TopicPartition(topic, partitionInfo.partition))
     }
 
-    private def getPartitionsToReset(groupId: String): Iterable[TopicPartition] = {
+    private def getPartitionsToReset(groupId: String): Seq[TopicPartition] = {
+      val allTopicPartitions = adminClient.listGroupOffsets(groupId).keys.toSeq
       if (opts.options.has(opts.allTopicsOpt)) {
-        adminClient.listGroupOffsets(groupId).keys
+        allTopicPartitions
       } else if (opts.options.has(opts.topicOpt)) {
         val topics = opts.options.valuesOf(opts.topicOpt).asScala
-        parseTopicPartitionsToReset(topics)
+        val topicPartitions = parseTopicPartitionsToReset(topics)
+        allTopicPartitions.filter(topicPartition => topicPartitions.contains(topicPartition))
```

Comment:
Yes, I think it is a reasonable feature to implement here and users might find it a little surprising if they couldn't do it.